src/controllers/authController.js

```diff
@@ -54,3 +54,51 @@ exports.registerNewUser = (req, res) => {
 		}
 	);
 };
+
+exports.login = async (req, res) => {
+	try {
+		const prevUser = await user.findOne({ username: req.body.username });
+
+		if (prevUser) {
+			const prevUserEmail = await bcrypt.compare(
+				req.body.email,
+				prevUser.email
+			);
+
+			if (prevUserEmail && req.body.email === prevUser.email) {
```

Comment:
please revise this `bcrypt.compare` is meant to compare a hashed string with a plain version. The user's password has been saved as a hashed string to the database, hence you should be comparing the password sent in the req.body with the user's password. as `bcrypt.compare(req.body.password, user.password)`